### ISSUE TITLE
FlutterErrorDetails now uses DiagnosticsNode context

### ISSUE THREAD

**Issue by droidery** [dnfield/flutter_svg#157]
On Flutter master, the context when using `FlutterErrorDetails` is now a `DiagnosticsNode` object instead of a string. Build of this package fails if you are on master with error:
```
file:///Users/droidery/.pub-cache/hosted/pub.dartlang.org/flutter_svg-0.12.4/lib/svg.dart:158:18: Error: The argument type 'String' can't be assigned to the parameter type 'DiagnosticsNode'.
 - 'DiagnosticsNode' is from 'package:flutter/src/foundation/diagnostics.dart' ('file:///Users/droidery/flutter/packages/flutter/lib/src/foundation/diagnostics.dart').
Try changing the type of the parameter, or casting the argument to 'DiagnosticsNode'.
        context: 'picture failed to precache',
                 ^
```

Ref: https://github.com/flutter/flutter/pull/30983

**Comment by dnfield** [comment]
This is a problem.  I've really hoped to avoid maintaining forks of this package for both master and stable, but this might require it.

@jacob314 , is there any work around for this?  If I want to support both Flutter stable and Flutter master, do I need to just get rid of the context parameter?

**Comment by jacob314** [comment]
To ease the migration we could change the type of the context property to Object and accept both String and DiagnosticsNode values for a few months.

**Comment by jacob314** [comment]
Unfortunately, flutter_svg also uses informationCollector which my CL also cleans up the API for in a way that can't be papered over with just a cleaned up type.

**Comment by jacob314** [comment]
I've created a pull request that refactors the error messages in flutter_svg. This would require a separate branch to support flutter_master.
https://github.com/dnfield/flutter_svg/pull/158

**Comment by dnfield** [comment]
Thanks!  I'll have to figure out how to handle this.  There are a few things that would benefit from a new ask constraint anyway.

**Comment by dnfield** [comment]
This is fixed in v0.13.0 with a new SDK version constraint.

**Comment by dnfield** [comment]
Thanks to @jacob314 for providing the new implementation!
